### DISCUSSION TITLE
Quartz job triggering multiple times in kubernates pods (nodes) with mongodb and C#

### DISCUSSION THREAD

**Discussion by MohamedSahir** [quartznet/quartznet#1991] [Q&A]
currently we are using asp.net core with rabbitmq and mongdb as database . we are using quartz job for data migration. but we observed jobs are triggred 5 times in 5 nodes.

i could see some solution before from quartz with ado.net store and sql server to lock the job store. based on this 
https://github.com/quartz-scheduler/quartz/issues/29.

could you please provide some useful documentation for mongodb jobstore lock and properties["quartz.jobStore.acquireTriggersWithinLock"] = "true";
this is the code i found for sql server but how can i modify for mongodb,is this solution solve the problem or any other suggested way ?
```     
        var properties = new NameValueCollection();
        properties["quartz.jobStore.type"] = "Quartz.Impl.AdoJobStore.JobStoreTX, Quartz";
        properties["quartz.jobStore.dataSource"] = "default";
        properties["quartz.jobStore.clustered"] = "true";
        properties["quartz.jobStore.selectWithLockSQL"] = string.Format(System.Globalization.CultureInfo.InvariantCulture, "SELECT * FROM {0}{1} WHERE {2} = {3} AND {4} = @lockName", new object[] { "{0}", "LOCKS", "SCHED_NAME", "{1}", "LOCK_NAME" });
        properties["quartz.jobStore.acquireTriggersWithinLock"] = "true";
        //properties["quartz.scheduler.instanceId"] = "AUTO";
        //properties["quartz.threadPool.threadCount"] = "1";
        //properties["quartz.jobStore.tablePrefix"] = "Scheduling.";
        //properties["quartz.dataSource.default.connectionString"] = @"Server = (local)\sqlexpress; Database = DB; Integrated Security = True";
        //properties["quartz.dataSource.default.provider"] = "SqlServer-20";
```

**Comment by lahma** [comment]
We don't have MongoDB support built here so hard to guess what might be the problem.
